packages/backend/src/graphql/resolvers/quote.ts

```diff
@@ -32,8 +33,15 @@ export const getQuote: QueryResolvers<ApolloContext>['quote'] = async (
 export const createQuote: MutationResolvers<ApolloContext>['createQuote'] =
   async (parent, args, ctx): Promise<ResolversTypes['QuoteResponse']> => {
     const quoteService = await ctx.container.use('quoteService')
+    const options: CreateQuoteOptions = {
```

Comment:
Because graphql doesn't do mutually exclusive types (:disappointed:), I think we need to move the old check from the quote service here
https://github.com/interledger/rafiki/blob/4db6f8e8ee6e4812af452af379855e4c3ff8d2f6/packages/backend/src/open_payments/quote/service.ts#L76-L78
```suggestion
    if (args.input.sendAmount && args.input.receiveAmount) {
      return {
        code: errorToCode[QuoteError.InvalidAmount].toString(),
        success: false,
        message: errorToMessage[QuoteError.InvalidAmount]
      }
    }
    const options: CreateQuoteOptions = {
```

Reply:
Initially I was looking at the `@oneOf` directive ([RFC](https://github.com/graphql/graphql-spec/pull/825)). Even though it's available in a couple of packages, is still a feature from the future and the behavior could change slightly. It would be nice if we can enforce the types at the GraphQL schema later on.

While playing around with it, I got to this point:
```ts
// schema.graphql

input QuoteAmount @oneOf {
  sendAmount: AmountInput
  receiveAmount: AmountInput
}

input CreateQuoteInput {
  paymentPointerId: String!
  amount: QuoteAmount
  receiver: String!
}
```
```ts
// graphql.generated.ts

export type QuoteAmount =
  { sendAmount: AmountInput; receiveAmount?: never; }
  |  { sendAmount?: never; receiveAmount: AmountInput; };

export type CreateQuoteInput = {
  paymentPointerId: Scalars['String'];
  amount?: InputMaybe<QuoteAmount>;
  receiver: Scalars['String'];
};
```

I _think_ it partially fulfills our needs for the `CreateQuoteInput`. 🤔